src/main/java/com/google/edith/servlets/UserInsights.java

```diff
@@ -33,61 +35,64 @@
 
   private String userId;
 
-  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();  
-  private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+  private static final DatastoreService datastore = 
+      DatastoreServiceFactory.getDatastoreService();  
+  private static final DateTimeFormatter dateFormatter = 
+      DateTimeFormatter.ofPattern("yyyy-MM-dd");
   private static final Gson gson = new Gson();
 
-  /** Assumes dates are in the form yyyy-MM-dd */
-  private static final Comparator<Key> SORT_BY_DATE = (Key item1, Key item2) -> {
-      try {
-        return ((String) (datastore.get(item1).getProperty("date")))
-            .compareTo((String) (datastore.get(item2).getProperty("date")));
-      } catch(EntityNotFoundException | NullPointerException e )  {
-        return 0;
-      }
-  };
-  
   public UserInsights(String userId) {
     this.userId = userId;
   }
 
   /**
-   * Finds the UserStats entity corresponding to {@code userId} in datastore.
-   * @return UserStats entity for this user
+   * Retrieves the Item entities corresponding to {@code userId} in datastore.
+   * @return List of Item entities for this user
    */
+  private List<Entity> retreiveUserItems() {
+    Filter idFilter = new FilterPredicate("userId", 
+                                FilterOperator.EQUAL,
+                                userId);
+    Query query = new Query("Item").setFilter(idFilter)
+                                    .addSort("date", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    return results.asList(FetchOptions.Builder.withLimit(Integer.MAX_VALUE));
+  }
+
+  /**
+   * Retrieves the UserStats entitie corresponding to {@code userId} in datastore.
+   * @return UserStats entity for this user
+   */ 
   private Entity retreiveUserStats() {
-    Filter idFilter = new FilterPredicate("UserId", 
+    Filter idFilter = new FilterPredicate("userId", 
                                 FilterOperator.EQUAL,
                                 userId);
     Query query = new Query("UserStats").setFilter(idFilter);
     PreparedQuery results = datastore.prepare(query);
     return results.asList(FetchOptions.Builder.withLimit(10)).get(0);
   }
 
-  /** This should only be called each time a new user makes an accout. */
+  /** 
+   * This should only be called each time a new user makes an accout
+   * and uploads a new receipt for the first time. 
+   */
   public void createUserStats() { 
-     List<Key> items = new ArrayList<>();
-
      Entity userStats = new Entity("UserStats");
-     userStats.setProperty("UserId", userId);
-     userStats.setProperty("Items", items);
+     userStats.setProperty("userId", userId);
+     userStats.setProperty("weekEndDates", new HashSet<String>());
+     userStats.setProperty("weeklyTotals", new ArrayList<String>());
```

Comment:
I created this as a way to keep track of weekly total spending, but there's no guarantee that the set will be in order, so I'm thinking of creating a list of strings with each element being in the form "weekEndDate = amount".
So one would like "06-20-2020 = 58.75"